clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -179,7 +179,7 @@ def err_verify_invalid_no_diags : Error<
     "%select{expected|'expected-no-diagnostics'}0 directive cannot follow "
     "%select{'expected-no-diagnostics' directive|other expected directives}0">;
```

Comment:
This looks like it needs templating?

Reply:
Can you please elaborate on this.

Reply:
I think @jrtc27 is saying that this diagnostic should also change to use `%0` similar to what you've done for `err_verify_no_directives` because it has the same issue with saying `'expected-no-diagnostics'`.

Reply:
I see, that can be done easily but I think according to LLVM PR standards, its best to limit one PR for one issue. Maybe you can start it as a different issue and then I will submit a PR for the same.

Reply:
I think this is the same issue -- in both cases, it's a hard-coded `expected-no-diagnostics` that is incorrect in the presence of a different prefix being used. Because of that, I'd still recommend doing it as part of this PR instead of as a separate one.

Reply:
Okay, sure thing, I will drop the changes within a few hours then.

Reply:
@AaronBallman, I don't think it needs templating. After a careful look I believe that the current implementation is for ```-verify``` only. Also the error says ```err_verify_invalid_no_diags``` which implicitly means that we are passing no ```-verify``` diagnostics in the first place. My point being in this case, there are no prefixes being used.

Reply:
@AaronBallman , please correct me if I am wrong.

Reply:
@Sh0g0-1758 Please have patience and don't ping people more often than once a wekk. Clang is very constrained on reviewer time of experienced people like Aaron.

Reply:
Ah really sorry about that. Since this was pending for a few days and I wanted to push more PRs for the other issues without making a separate branch and rebasing them, I got a little impatient. Will surely keep this thing in mind from now on.

Reply:
I definitely sympathize with you wanting to push more than one PR at a time, but the way to do it is to follow GitHub flow, specifically a part of it that says you should work in branches, and create your PRs from then: https://docs.github.com/en/get-started/using-github/github-flow#create-a-branch. This way PRs that touch different code can go through as many rounds of review as needed, without blocking other PRs.

Nagging reviewers won't get you anywhere, or even worse, deter reviewers from your PRs.

Reply:
Yes I understand. I made a beginners mistake of not making a branch for the changes in this PR. Will surely keep that in mind from now on. Also I hope you can overlook my enthu, was not trying to nag any reviewers.

Reply:
No worries about the extra pings (we'd rather see people "too" enthusiastic than disengaged!), just something to keep in mind for the future. :-)

The reason I think this diagnostic also needs to be updated is because of: https://godbolt.org/z/sEf4o8YY9 and https://godbolt.org/z/8oMWffoxY where it says `'expected-no-diagnostics'` in the diagnostic message. I think those should also say `'foo-no-diagnostics'` similar to how you've fixed it for the `no expected directives found` warning.

Reply:
Thanks a lot for looking over it. 

I have made the required changes and added more tests for the same. Please take a look and suggest changes.